libs/components/modals/testing/src/confirm/confirm-testing.service.ts

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable @nx/enforce-module-boundaries */
+import {
+  SkyConfirmCloseEventArgs,
+  SkyConfirmConfig,
+  SkyConfirmInstance,
+  SkyConfirmServiceInterface,
+} from '@skyux/modals';
+
+import { SkyConfirmTestingController } from './confirm-testing.controller';
+
+interface TestSubject {
+  config: SkyConfirmConfig;
+  instance: SkyConfirmInstance;
+}
+
+function assertConfirmOpen(
+  value: TestSubject | undefined,
+): asserts value is TestSubject {
+  if (value === undefined) {
+    throw new Error('A confirm dialog is expected to be open but is closed.');
+  }
+
+  return;
+}
+
+function assertConfirmClosed(
+  value: TestSubject | undefined,
+): asserts value is undefined {
+  if (value !== undefined) {
+    throw new Error('A confirm dialog is expected to be closed but is open.');
+  }
+
+  return;
+}
+
+/**
+ * @internal
+ */
+export class SkyConfirmTestingService
+  implements SkyConfirmServiceInterface, SkyConfirmTestingController
```

Comment:
You're implementing a class like it's an interface, but classes should be extended.